forex_python/converter.py

```diff
@@ -79,35 +87,61 @@ def get_rate(self, base_cur, dest_cur, date_obj=None):
         raise RatesNotAvailableError("Currency Rates Source Not Ready")
 
     def convert(self, base_cur, dest_cur, amount, date_obj=None):
+        use_decimal = False
         if isinstance(amount, Decimal):
             use_decimal = True
-        else:
-            use_decimal = self._force_decimal
+        elif self._force_decimal:
+            raise DecimalFloatMismatchError("Decimal is forced. Amount must be Decimal")
 
         if base_cur == dest_cur:  # Return same amount if both base_cur, dest_cur are same
-            if use_decimal:
-                return Decimal(amount)
-            return float(amount)
+            return amount
+        
+        rate = self.get_rate(base_cur, dest_cur, date_obj=date_obj, use_decimal=use_decimal)
 
+        return rate * amount
+
+class CurrencyRatesForexAPI(CurrencyRatesBase):
+    def _source_url(self):
+        return "https://theforexapi.com/api/"
+
+class CurrencyRatesExchangeRateHost(CurrencyRatesBase):
+    def _source_url(self):
+        return "https://api.exchangerate.host/"
```

Comment:
These intermediate classes allow for defining multiple providers, and overriding the implementation slightly for each one